docs/src/main/asciidoc/spanner.adoc

```diff
@@ -0,0 +1,190 @@
+The `SpannerDialect` produces SQL, DML, and DDL statements for most common entity types and relationships using standard Hibernate and Java Persistence annotations.
+
+Please see the following sections for important details and differences due to Cloud Spanner's features and data model.
+
+=== Quick Set-Up
+
+Maven coordinates for the dialect:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-spanner-hibernate-dialect</artifactId>
+  <version> TBD </version>
+</dependency>
+----
+
+Maven coordinates for the official Cloud Spanner Driver:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-spanner</artifactId>
+  <version> TBD </version>
+</dependency>
+----
+
+Configuring the `SpannerDialect` and a Cloud Spanner Driver class is typical of all Hibernate dialects in the `hibernate.properties` file:
+
+----
+hibernate.dialect=com.google.cloud.spanner.hibernate.SpannerDialect
+hibernate.connection.driver_class=com.google.cloud.spanner.jdbc.JdbcDriver
+hibernate.connection.url=jdbc:cloudspanner:/projects/{INSERT_PROJECT_ID}/instances/{INSERT_INSTANCE_ID}/databases/{INSERT_DATABASE_ID}
+----
+
+The https://cloud.google.com/docs/authentication/getting-started[service account JSON credentials] file's location should be in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+The driver will use default credentials set in the Google Cloud SDK `gcloud` application otherwise.
+
+The dialect and driver are compatible with all values (`create`, `create-drop`, and `update`) of the `hibernate.hbm2ddl.auto` setting.
+
+
+=== Spanner Dialect
+
+The `SpannerDialect` supports most of the standard Hibernate and Java Persistence annotations but there are important differences in features because of differences in https://cloud.google.com/spanner/docs/schema-and-data-model[Cloud Spanner's data model] from traditional SQL databases.
+
+These are unsupported features in the Cloud Spanner Hibernate Dialect.
+Please see the following sections for details:
+[options="header"]
+|===
+| Unsupported Feature | Description
+| Constraints | No support for `FOREIGN KEY`, `UNIQUE`, or `ON DELETE CASCADE`. Cloud Spanner does not enforce relationship links outside of Hibernate.
+| Catalog and schema in table names | Tables names cannot contain periods or other punctuation.
+| Column default values | Cloud Spanner treats `null` like any other value and it gets no special handling.
+| Big-decimal or arbitrary-precision numbers | Cloud Spanner does not support `java.sql` types of `NUMERIC` or `DECIMAL`
+|===
+
+==== Relationships
+
+The dialect supports all of the standard relational annotations:
+
+- `@OneToOne`
+- `@OneToMany`
+- `@ManyToOne`
+- `@ManyToMany`
+
+These can be used via `@JoinTable` or `@JoinColumn`.
+However, because Cloud Spanner does not support Foreign Keys, foreign-key-columns are just regular columns in Cloud Spanner.
```

Comment:
I think it's worth emphasizing that Spanner does support foreign keys, it just doesn't support foreign key **constraints**.
```suggestion
However, because Cloud Spanner does not support foreign key constraints, foreign-key-columns are just regular columns in Cloud Spanner.
```